app/react/src/client/preview/render.tsx

```diff
@@ -28,9 +37,51 @@ export const render: ArgsStoryFn<ReactFramework> = (args, context) => {
 
 const renderElement = async (node: ReactElement, el: Element) =>
   new Promise((resolve) => {
-    ReactDOM.render(node, el, () => resolve(null));
+    // Create Root Element conditionally for new React 18 Root Api
+    const root = getReactRoot(el);
+
+    if (root) {
+      root.render(
+        <CallbackWrapper
+          callback={() => {
+            resolve(null);
+          }}
+        >
+          {node}
+        </CallbackWrapper>
+      );
+    } else {
+      ReactDOM.render(node, el, () => resolve(null));
+    }
   });
 
+const unmountElement = (el: Element) => {
+  const root = nodes.get(el);
+  if (root) {
```

Comment:
can we also check for `FRAMEWORK_OPTIONS?.newReactRootApi` to make this more robust? otherwise we can possibly get burned by race conditions? 🤷‍♂️